features/step_definitions/search_steps.rb

```diff
@@ -34,28 +34,27 @@
 end
 
 Then /^search analytics for "(.*)" are reported$/ do |term|
-  found = false
   sought = "dp=#{CGI::escape("/search/all?keywords=#{term.sub(' ', '+')}")}"
-  $proxy.har.entries.each do |e|
-    found = true if e.request.url.include? sought
-  end
-  expect(found).to be(true)
+  wait_until { proxy_has_request_containing sought }
+  expect(proxy_has_request_containing sought).to be(true)
```

Comment:
This will raise a timeout error rather than explaining what happened. Would it be useful to rescue the error and raise an expectation type error so when it fails it'll be clearer why?

Reply:
I think it's fine, because the exception will point to the `wait_until` line which has the condition.